### DISCUSSION TITLE
Welcome to Project Minty

### DISCUSSION THREAD

**Discussion by mjehrhart** [mjehrhart/minty#1] [General]
Welcome to the discussion forum. Please feel free to comment if you want.

**Comment by omac777** [comment]
Awesome work wielding egui/eframe to do something very useful.  I'm still trying to wrap my head as to how your your main.rs invokes the gui and decides what directory from which to start finding files files from.  It doesn't seem to be an argument from the cli.
I got it:
1)It goes from the main to the controller
https://github.com/mjehrhart/minty/blob/b352762bc6fd913fb1caba24923c2ec244a76b2d/src/main.rs#L22
2)the controller creates the top_layout
https://github.com/mjehrhart/minty/blob/b352762bc6fd913fb1caba24923c2ec244a76b2d/src/app/controller.rs#L252
3)the top_layout creates the top_panel
https://github.com/mjehrhart/minty/blob/b352762bc6fd913fb1caba24923c2ec244a76b2d/src/app/top_panel.rs#L10
4)within the top_panel, it calls return_dfer2 to populate collection, d2 and finder
https://github.com/mjehrhart/minty/blob/b352762bc6fd913fb1caba24923c2ec244a76b2d/src/app/top_panel.rs#L91
5)return_dfer2 is found in main:
https://github.com/mjehrhart/minty/blob/b352762bc6fd913fb1caba24923c2ec244a76b2d/src/main.rs#L25

Phew my poor head LOL.  I'm guessing this is a "Model View Controller" design since you mentioned controller as the name of your source file.

The other aspect that's interesting is you have 2 tables, one for the list of files, another for the selected files.
The selected files table is the one that has the delete button that deletes the file from the os and from the collection list of files returned from your finder that uses rayon/jwalk.  That implies the model of data you use for both table gui components was "collection".  I can tell you reused your efforts from your **ffind** project to create minty.

A lot of effort and understanding into rust and egui/eframe went into this.  You really crystallized a lot of the usage of both and thank you for that.  You're a pathfinder.  You deserve many more stars than what you have IMHO.  Cheers!
